app/components/schools/participants/coc_status_table.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+module Schools
+  module Participants
+    class CocStatusTable < BaseComponent
+      def initialize(induction_records:)
+        @induction_records = induction_records
+      end
+
+      def ineligible_participants?
+        ParticipantProfile::ECF.ineligible_status.where(id: induction_records.map(&:participant_profile_id)).any?
+      end
+
+      def transferring_out_participants?
+        induction_records.any?(&:transferring_out?)
+      end
+
+      def transferring_in_participants?
+        induction_records.any?(&:transferring_in?)
+      end
+
+      def transferred_participants?
+        induction_records.any?(&:transferred?)
+      end
+
+      # this doesn't really work in the new data model as there could be multiple
+      # programmes running in the cohort
+      def school_chose_cip?
```

Comment:
Could we get rid of this then instead of leaving an edge case ready to break in the future? We could render "n/a" for the materials supplier column for non-cip participants?